asteroid/dsp/overlap_add.py

```diff
@@ -146,3 +147,61 @@ def _reorder_sources(
     _, perms = PITLossWrapper.find_best_perm(pw_losses, n_src)
     current = PITLossWrapper.reorder_source(current, n_src, perms)
     return current.reshape(batch, frames)
+
+
+class DualPathProcessing(nn.Module):
+    def __init__(self, chunk_size, hop_size):
+        super(DualPathProcessing, self).__init__()
+        self.chunk_size = chunk_size
+        self.hop_size = hop_size
+        self.n_orig_frames = None
+
+    def unfold(self, x):
+        # x is (batch, chan, frames)
+        batch, chan, frames = x.size()
+        assert x.ndim == 3
+        self.n_orig_frames = x.shape[-1]
+        unfolded = torch.nn.functional.unfold(
+            x.unsqueeze(-1),
+            kernel_size=(self.chunk_size, 1),
+            padding=(self.chunk_size, 0),
+            stride=(self.hop_size, 1),
+        )
+
+        return unfolded.reshape(
+            batch, chan, self.chunk_size, -1
+        )  # (batch, chan, chunk_size, n_chunks)
+
+    def fold(self, x, output_size=None):
+        output_size = output_size if output_size is not None else self.n_orig_frames
```

Comment:
This is what I would do instead of the assert in `unfold`.
If you have `DualPathProcessing` in your model, you would have to instantiate it everytime, it doesn't seem good to me.

Reply:
Ok I see. No need for assert because code will fail lol